### PR TITLE
bug 2039670: Skip deployment of PDBs on the SNO topology

### PR DESCRIPTION
**Pull request by ingvagabund**
Builds on top of https://github.com/openshift/library-go/pull/1246 and WithIsSNOCheck from https://github.com/openshift/library-go/pull/1238.

### CODE REVIEW

**Review by bertinatto — COMMENTED** [review]
@ingvagabund I think we got this sorted out already. Please take a look: https://github.com/openshift/cluster-csi-snapshot-controller-operator/commit/af5cba613d63317d2fd3e1371b94c3acbff6e024

**Comment by ingvagabund** [comment]
@bertinatto of course you have /O\. I will just update the code so the PDBs are created conditionally. To avoid installation on the SNO topology.

**Comment by ingvagabund** [comment]
Once https://github.com/openshift/cluster-csi-snapshot-controller-operator/pull/110 gets merged I will rebase this PR.

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -73,14 +74,41 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
...
+			isSNO, precheckSucceeded, err := guard.IsSNOCheckFnc(configInformers.Config().V1().Infrastructures())()
+			if err != nil {
+				klog.Errorf("IsSNOCheckFnc failed: %v", err)
+				return false
+			}
+			if !precheckSucceeded {
+				klog.V(4).Infof("IsSNOCheckFnc precheck did not succeed, skipping")
+				return false
+			}
+			return !isSNO
```
Comment (jsafrane) [inline pkg/operator/starter.go:98]: Can you create `isSNO()` out of this and use `isSNO()` / `!isSNO` below?

Comment (ingvagabund) [reply pkg/operator/starter.go:98]: Unfortunately, the create/delete conditionals are not opposite of each other. They both can be false when:
- `configInformers.Config().V1().Infrastructures()`'s hasSynced is false
- retrieval of the `cluster` infrastructure object failed.

**Pull request merged**